apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -101,24 +110,57 @@ export const saveStudent = (studentId) => {
   };
 };
 
-// Adds a student, with the given row id (studentId), from an addRow or
-// a newStudentRow.
-export const addStudent = (studentId) => {
+// Saves all RowType.STUDENT currently being edited and adds all
+// RowType.NEW_STUDENT currently being edited.
+export const saveAllStudents = () => {
   return (dispatch, getState) => {
     const state = getState().manageStudents;
-    dispatch(startSavingStudent(studentId));
-    addStudentOnServer(state.editingData[studentId], state.sectionId, (error, data) => {
+
+    // Currently, every update is an individual call to the server.
+    const currentlyEditedData = convertStudentDataToArray(state.editingData);
+    let studentsToSave = currentlyEditedData.filter(student => student.rowType === RowType.STUDENT);
+    for (let i = 0; i<studentsToSave.length; i++) {
+      dispatch(saveStudent(studentsToSave[i].id));
+    }
+
+    // Adding students can be saved together.
+    const arrayOfEditedData = convertStudentDataToArray(state.editingData);
+    const newStudentsToAdd = arrayOfEditedData.filter(student => student.rowType === RowType.NEW_STUDENT).map(
+      student => student.id);
```

Comment:
Style nit: This is a long line, I'd break it across your chained method calls.
```js
const newStudentsToAdd = arrayOfEditedData
  .filter(student => student.rowType === RowType.NEW_STUDENT)
  .map(student => student.id);
```
...also
![combo_super_combo_finish](https://user-images.githubusercontent.com/1615761/36917405-9c68732e-1e0b-11e8-94e9-089f49ddd3bb.jpg)